xwiki-rendering-test/src/main/java/org/xwiki/rendering/test/cts/TestData.java

```diff
@@ -139,15 +139,12 @@ public String computeTestName()
     @Override
     public boolean equals(Object object)
     {
-        if (object == null) {
+        if (object == null && object.getClass() != getClass()) {
```

Comment:
Since TestData is not really designed to be extended you would replace this whole if by `if (!(object instanceof TestData))`.

Also I would move it after the `if (object == this)`.

Reply:
okay

Reply:
Done.